orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy

```diff
@@ -358,7 +358,6 @@ class TaskController {
         reason
       )
     }
-    executionRepository.updateStatus(PIPELINE, id, ExecutionStatus.CANCELED)
```

Comment:
turns out this was the main cause of my melted brain - queueing a CancelExecution message but also monkeying with the underlying state directly is a bad combo

I suspect the side-effect of this is that it will take longer for the pipeline to reflect a CANCELED state in the UI as its not updated to that until it is actually done - maybe there is a use case here for creating that artificial state (essentially it is running with a different label) but we do already have a  cancelled boolean flag on the execution so we can I think get by without introducing another execution status.

Reply:
I seem to recall trying to argue someone out of making that change but being brain fried at the time and not being able to articulate my concerns very well.

Reply:
I just wonder if having a new state is a better thing than having state + boolean combo.